src/main/java/bio/terra/service/JobService.java

```diff
@@ -189,7 +192,21 @@ public JobModel mapFlightStateToJobModel(FlightState flightState) {
     }
 
     public JobModel retrieveJob(String jobId, AuthenticatedUserRequest userReq) {
-        stairway.verifyListFlightAccess(jobId, buildUserRequestInfo(userReq));
+        boolean canListAnyJob = true;
+        if (userReq != null) {
+            // currently, this check will be true for stewards only
+            canListAnyJob = samService.isAuthorized(
+                userReq,
+                SamClientService.ResourceType.DATAREPO,
+                appConfig.getResourceId(),
+                SamClientService.DataRepoAction.LIST_JOBS);
+        }
```

Comment:
If `userReq` is null then we proceed with steward-level permissions. Should this throw an exception if userReq is null?

Reply:
Rori and I discussed that and concluded that this null check seems to be for testing only. I would be in favor of throwing an exception if userReq is null, but then I think we'd have to make another change so that the tests can bypass the SAM check. Previously, this wasn't an issue because the SAM check was in the controller, and the tests just hit the JobService. Now we've moved the SAM check to the JobService, so the tests can no longer bypass it. Maybe we can change the tests to set another property of the userReq to null to indicate that the SAM check should be skipped -- say the token?

Reply:
Ah I see, that makes sense. We stub out SAM in our connected tests so that calls don't actually go to dev SAM server. What do you think about minting fake AuthUserRequests in the tests to pass in? I think this could allow us to avoid putting test-specific code outside of the tests.

Reply:
Sounds good. I'll look into this and discuss with Rori this afternoon. Thanks for the recommendation.

Reply:
Per conversation at stand-up, moving this work to a separate JIRA ticket. We will resolve it in another branch/PR.